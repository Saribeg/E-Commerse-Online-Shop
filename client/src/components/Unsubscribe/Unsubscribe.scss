@import '../../scss/resources/resources';

.section-unsubscribe {
  padding: 65px 0;
}

.unsubscribe {
  @include flex-center-between;

  @include max-width($landscape-tablet) {
    flex-direction: column;
  }

  &-form {
    display: flex;
    justify-content: center;
    width: 100%;

    @include max-width($landscape-tablet) {
      flex-direction: column;
      margin-bottom: 10px;
      justify-content: center;
      align-items: center;
    }

  }

  &-email {
    border: none;
    outline: none;
    max-width: 318px;
    width: 100%;
    border-radius: 22px;
    padding: 15px;
    background-color: $grey-light;
    margin-right: 39px;

    @include max-width($landscape-tablet) {
      margin-right: 0;
      margin-bottom: 10px;
    }
  }

  &-btn {
    max-width: 300px;
    width: 100%;
    padding: 12px 0;
    border-radius: 100px;
    background-color: $black;

    font-family: $font-openSans;
    font-size: 13px;
    font-weight: bold;
    text-align: center;
    letter-spacing: 1.4px;
    text-transform: uppercase;
    color: #ffffff;
    border: none;
    outline: none;
    cursor: pointer;

  }
}