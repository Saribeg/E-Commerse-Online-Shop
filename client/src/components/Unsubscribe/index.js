import React, {Component} from "react";
import './Unsubscribe.scss';
import axios from "axios";

class Unsubscribe extends Component {
    state = {
        submitted: false
    };

    unsubscribeMail = (e) => {
        e.preventDefault();
        axios.post(window.location.href);
        this.setState({submitted: true});
    };

    render() {

        return (
            <>
                <section className="section-unsubscribe">
                    <div className="container">
                        <div className="unsubscribe">
                            <form onSubmit={this.unsubscribeMail} className="unsubscribe-form">
                                <input id='unsubscribe-btn' type="submit" className="unsubscribe-btn"
                                       value='UNSUBSCRIBE'/>
                            </form>
                        </div>
                    </div>
                </section>
            </>
        );
    }
}

export default Unsubscribe;
