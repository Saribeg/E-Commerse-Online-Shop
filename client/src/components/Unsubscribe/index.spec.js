import React from 'react';
import Unsubscribe from './index';

describe('Unsubscribe component', () => {
    it('Unsubscribe component render', () => {
        const wrapper  = shallow(<Unsubscribe />)
        expect(wrapper).toMatchSnapshot()
    })
})