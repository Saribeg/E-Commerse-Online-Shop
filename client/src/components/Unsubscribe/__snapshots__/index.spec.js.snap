// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unsubscribe component Unsubscribe component render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unsubscribe />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <section
        className="section-unsubscribe"
      >
        <div
          className="container"
        >
          <div
            className="unsubscribe"
          >
            <form
              className="unsubscribe-form"
              onSubmit={[Function]}
            >
              <input
                className="unsubscribe-form-btn"
                id="unsubscribe-btn"
                type="submit"
                value="UNSUBSCRIBE"
              />
            </form>
          </div>
        </div>
      </section>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="unsubscribe"
          >
            <form
              className="unsubscribe-form"
              onSubmit={[Function]}
            >
              <input
                className="unsubscribe-form-btn"
                id="unsubscribe-btn"
                type="submit"
                value="UNSUBSCRIBE"
              />
            </form>
          </div>
        </div>,
        "className": "section-unsubscribe",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="unsubscribe"
          >
            <form
              className="unsubscribe-form"
              onSubmit={[Function]}
            >
              <input
                className="unsubscribe-form-btn"
                id="unsubscribe-btn"
                type="submit"
                value="UNSUBSCRIBE"
              />
            </form>
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form
              className="unsubscribe-form"
              onSubmit={[Function]}
            >
              <input
                className="unsubscribe-form-btn"
                id="unsubscribe-btn"
                type="submit"
                value="UNSUBSCRIBE"
              />
            </form>,
            "className": "unsubscribe",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className="unsubscribe-form-btn"
                id="unsubscribe-btn"
                type="submit"
                value="UNSUBSCRIBE"
              />,
              "className": "unsubscribe-form",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "unsubscribe-form-btn",
                "id": "unsubscribe-btn",
                "type": "submit",
                "value": "UNSUBSCRIBE",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "form",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "section",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <section
          className="section-unsubscribe"
        >
          <div
            className="container"
          >
            <div
              className="unsubscribe"
            >
              <form
                className="unsubscribe-form"
                onSubmit={[Function]}
              >
                <input
                  className="unsubscribe-form-btn"
                  id="unsubscribe-btn"
                  type="submit"
                  value="UNSUBSCRIBE"
                />
              </form>
            </div>
          </div>
        </section>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <div
              className="unsubscribe"
            >
              <form
                className="unsubscribe-form"
                onSubmit={[Function]}
              >
                <input
                  className="unsubscribe-form-btn"
                  id="unsubscribe-btn"
                  type="submit"
                  value="UNSUBSCRIBE"
                />
              </form>
            </div>
          </div>,
          "className": "section-unsubscribe",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="unsubscribe"
            >
              <form
                className="unsubscribe-form"
                onSubmit={[Function]}
              >
                <input
                  className="unsubscribe-form-btn"
                  id="unsubscribe-btn"
                  type="submit"
                  value="UNSUBSCRIBE"
                />
              </form>
            </div>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <form
                className="unsubscribe-form"
                onSubmit={[Function]}
              >
                <input
                  className="unsubscribe-form-btn"
                  id="unsubscribe-btn"
                  type="submit"
                  value="UNSUBSCRIBE"
                />
              </form>,
              "className": "unsubscribe",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  className="unsubscribe-form-btn"
                  id="unsubscribe-btn"
                  type="submit"
                  value="UNSUBSCRIBE"
                />,
                "className": "unsubscribe-form",
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "unsubscribe-form-btn",
                  "id": "unsubscribe-btn",
                  "type": "submit",
                  "value": "UNSUBSCRIBE",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": "form",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "section",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
